demo-circuit/src/blaze_csw/data_structures.rs

```diff
@@ -0,0 +1,288 @@
+use algebra::ToConstraintField;
+use cctp_primitives::{
+    commitment_tree::{sidechain_tree_alive::FWT_MT_HEIGHT, CMT_MT_HEIGHT},
+    proving_system::verifier::ceased_sidechain_withdrawal::PHANTOM_CERT_DATA_HASH,
+    type_mapping::FieldElement,
+    utils::{
+        commitment_tree::DataAccumulator, data_structures::BackwardTransfer, get_bt_merkle_root,
+    },
+};
+use primitives::{FieldBasedHash, FieldBasedMerkleTreePath, FieldHasher};
+
+use crate::{
+    constants::constants::BoxType, type_mapping::*, GingerMHTBinaryPath, MST_MERKLE_TREE_HEIGHT,
+    PHANTOM_FIELD_ELEMENT, PHANTOM_SECRET_KEY_BITS, SC_CUSTOM_HASH_LENGTH, SC_PUBLIC_KEY_LENGTH,
+    SC_TX_HASH_LENGTH,
+};
+
+#[derive(Clone, Debug)]
+pub struct WithdrawalCertificateData {
+    pub ledger_id: FieldElement,
+    pub epoch_id: u32,
+    pub bt_root: FieldElement, // Merkle root hash of all BTs from the certificate (recall that MC hashes all complex proof_data params from the certificate)
+    pub quality: u64,
+    pub mcb_sc_txs_com: FieldElement,
+    pub ft_min_amount: u64,
+    pub btr_min_fee: u64,
+    pub custom_fields: Vec<FieldElement>,
+}
+
+impl WithdrawalCertificateData {
+    pub fn new(
+        ledger_id: FieldElement,
+        epoch_id: u32,
+        bt_list: Vec<BackwardTransfer>,
+        quality: u64,
+        mcb_sc_txs_com: FieldElement,
+        ft_min_amount: u64,
+        btr_min_fee: u64,
+        custom_fields: Vec<FieldElement>,
+    ) -> Self {
+        Self {
+            ledger_id,
+            epoch_id,
+            bt_root: get_bt_merkle_root(if bt_list.is_empty() {
+                None
+            } else {
+                Some(&bt_list)
+            })
+            .unwrap(),
+            quality,
+            mcb_sc_txs_com,
+            ft_min_amount,
+            btr_min_fee,
+            custom_fields,
+        }
+    }
+
+    pub(crate) fn get_phantom(num_custom_fields: u32) -> Self {
+        Self {
+            ledger_id: PHANTOM_FIELD_ELEMENT,
+            epoch_id: 0,
+            bt_root: get_bt_merkle_root(None).unwrap(),
+            quality: 0,
+            mcb_sc_txs_com: PHANTOM_FIELD_ELEMENT,
+            ft_min_amount: 0,
+            btr_min_fee: 0,
+            custom_fields: vec![PHANTOM_FIELD_ELEMENT; num_custom_fields as usize],
```

Comment:
Since we implemented the `WithdrawalCertificateData::get_default()` function (see commit [51b8680341bc3a4c82568422d4e6cff927a169d2](https://github.com/HorizenOfficial/zendoo-sc-cryptolib/commit/51b8680341bc3a4c82568422d4e6cff927a169d2)), what should be done with `WithdrawalCertificateData::get_phantom()`? Should it just call `get_default()`?

Reply:
Yes, it should be fine

Reply:
Done with commit [cd72c64b705c3fc92f5fe40a8e540bbdeb9d523f](https://github.com/HorizenOfficial/zendoo-sc-cryptolib/commit/cd72c64b705c3fc92f5fe40a8e540bbdeb9d523f).